x-pack/plugins/observability_solution/synthetics/public/apps/synthetics/components/settings/data_retention/common.test.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { formatAge } from './common';
+
+describe('formatAge', () => {
+  it('should return -- when age is not provided', () => {
+    expect(formatAge()).toBe('--');
+  });
+
+  it('should handle singular day count', () => {
+    expect(formatAge('1d')).toBe('1 day + rollover');
+  });
+
+  it('should return formatted age when age is provided', () => {
+    expect(formatAge('5d')).toBe('5 days + rollover');
```

Comment:
Should we care if there's no `d` on the age value?

Reply:
I don't think so. If there's no `d` it will display the output regardless. If there's a different char it won't work, but I think this API always gives out `day` spans.